app/models/teacher_set.rb

```diff
@@ -160,10 +160,11 @@ def self.for_query(params)
       sets = sets.where("set_type = ?", params[:type])
     end
 
-    [:language, :availability].each do |prop|
-      if !params[prop].nil? && params[prop].size > 0
-        sets = sets.where("#{prop} IN (?)", params[prop])
-      end
+    if !params[:language].nil? && params[:language].size > 0
```

Comment:
instead of !params[:language].nil?  we can use params[:language].present?

Reply:
done